django/core/validators.py

```diff
@@ -167,15 +169,21 @@ class EmailValidator:
         # literal form, ipv4 or ipv6 address (SMTP 4.1.3)
         r'\[([A-f0-9:.]+)\]\Z',
         re.IGNORECASE)
-    domain_whitelist = ['localhost']
+    domain_allowlist = ['localhost']
```

Comment:
Subclassing `EmailValidator`(validators in general) is a valid option. Even though this is not documented per se, I'd do a deprecation for `domain_whitelist` to since we are not changing it for technical/security reasons.

Reply:
Right, `domain_whitelist` should be left as a `@property` that warns and returns `self.domain_allowlist`.